src/app/forms/services/dynamic-form.service.ts

```diff
@@ -123,22 +123,25 @@ export class DynamicFormService {
     validators.forEach(v => control.addAsyncValidators(this.asyncValidatorMap[v.name](v.args)));
   }
 
-  static updateValidity(form: FormGroup | FormArray, errors: GlobalError[]) {
+  static validate(form: CustomFormGroup | CustomFormArray | FormGroup | FormArray, errors: GlobalError[], updateValidity = true) {
     Object.entries(form.controls).forEach(([, value]) => {
-      if (!(value instanceof CustomFormControl)) {
-        this.updateValidity(value as CustomFormGroup | CustomFormArray, errors);
+      if (!(value instanceof FormControl || value instanceof CustomFormControl)) {
+        this.validate(value as CustomFormGroup | CustomFormArray, errors, updateValidity);
```

Comment:
```suggestion
        this.validate(value, errors, updateValidity);
```
not sure if the casting is necessary

Reply:
it won't accept a type of AbstractControl as its missing formArray so I think it will need casting like this

Reply:
I tried this too, @owen-corrigan-bjss's right.